packages/ui-components/src/components/Pagination/Pagination.component.js

```diff
@@ -60,15 +60,18 @@ export const Pagination = ({
   }
 
   const handlePrevClick = (event) => {
-    if (controlTotalPage || pages || controlPage) {
-      setControlCurrentPage(controlPage > 1 ? controlPage - 1 : undefined)
+    if (controlPage && controlPage > 1) {
+      setControlCurrentPage(controlPage - 1)
     }
+
     onPressPrevious && onPressPrevious(event)
   }
 
   const handleNextClick = (event) => {
-    setControlCurrentPage(controlPage < controlTotalPage || controlPage ? controlPage + 1 : controlPage)
-
+    // set controlPage +1 if controlPage is not undefined and controlPage is less than controlTotalPage. Set conrolPage +1 also if controlTotalPage is undefined.
+    if (!controlTotalPage || (controlPage && controlPage < controlTotalPage)) {
```

Comment:
If `controlPage` and `controlTotalPage` are both falsy then this expression evaluates to `true` and you will try to increment the value of `controlPage` even though you shouldn't. The reason for this is that as soon as one segment of a boolean expression connected via `||` is true then javascript stops evaluating further and will treat the expression as true.

You'll need to change this so it really only does anything if `controlPage` is truthy. Something like:
```suggestion
    if (controlPage) {
      if ((controlPage < controlTotalPage) || !controlTotalPage) {
```
I would suggest separating it into two nested ifs for readability.  Also please add a test that tests this edge case (controlPage truthy, controlTotalPage falsy)